### ISSUE TITLE
[Doc] wiki/GUI:-Savestates needs updating... question?

### ISSUE THREAD

**Issue by giantclambake** [BlitterStudio/amiberry#1552]
TBH, I was going to just submit updates for this myself ... and I'll happily still do that ....however recent improvements wrt amiberry's handling of savestates & .uss files, raises more questions than answers...in as far as documentation goes....

1. {'caveat emptor' ;^} ... we've discussed .uss usage before, it's original intents/purposes...etc et_al. In recent times, when you floated the idea of mime-type association with .uss files, that had me raise an eyebrow, from the 'we want to actively promote & use these files now?' perspective....in the process of looking for a byte sequence to match, I noted the .uss file structure, referenced the original Amiga `<file>` loaded, and when I further discovered that `amiberry --statefile somesavestate.uss -G` worked as expected, we've ended up at `amiberry somesavestate.uss` direct-loading .... the amiberry manpage is up-to-date but the wiki here needs updating in these regards....
2. On top of that, I noted that you noticed I was using whdload.lha files and the whdbooter process, to create savestates in that emulation scenario...I was half expecting you to quip 'you should not use savestates in that way' ;) Fact is I've had more than several hours, to test more than 50 Retroplay whdload.lha archives, and create/update/(invalidate ;) savestate .uss files for these titles, both with associated config.uae files and without (use the XML only)... and had zero failures ; the system seems to be just as reliable using the whdload.lha/whdbooter emulation environment, as it does wrt using floppy disk images in emulation.
3. The words "**There are a total of 15 available Save State slots**" can give the impression that the user can only have 15 savestates ~ it is not the case (anymore?) --- for example, 'Slot 0' may have any number of savestates (*.uss files) _of a different name_ ....this distinction should be pointed out...ie; the concept of 'Slot 0' being the starting level for every game .uss file, or group game genres by slot # etc etc....I'm sure some folks would like to know about this... =)
4. Yes, I did use the word 'invalidate'...the typical example would be changing something in the config.uae file, that affects the expected emulation state -- any .uss file, will start as it did before any changes were made to the config.uae file...thus the savestate file would be considered invalid. There are other nuances of savestate handling, when it comes to the whdload + whdbooter + config.uae + savestate.uss scenario ...ie; the worst of this was fixed via #1546 (thanks ;), but notwithstanding, changing the emulation config, and changing the whdload/whdbooter options stored in that config.uae, are somewhat mutually exclusive...ie; you can load/change/save the given config.uae to affect the emulation ~ you have to load the actual whdload.lha, to make any changes related to whdload/whdbooter options, before you save these changes to the given config.uae ...(as said in the ticket, I don't mind this ;) ...more fodder for documentation here...
5. The GUI helptext needs updating as well, but it's actually leading the way here, by at least indicating savestates should never be used in the case of the emulation engine utilizing JIT/(PPC)/RTG .... in testing here, 100% failure rate with these scenarios 8) Such should be mentioned on this wiki page as well...along with the fact that CDTV/CD32 machine types won't work either....(I believe with these modes, the emulation state 'loses' the CD drive entity)... 
6. Contrary to the Savestate panel helptext, I find the use of Savestates wrt HDD setups (dirs or .hdf), to in practice be as reliable as other emulation environments mentioned above. For example, I can launch straight into a running DPaint-IV instance, from .uss file, even though the machine makeup is an '030 A3000, wb2.11 .hdf (from AF10), with DPaint and others installed in a directory mount, pointing to DH1: (Work:) ... seems to work fine...but this usage shouldn't be 'lumped' into the same scenario as JIT/RTG -- that will always fail (in my testing) ; HDD usage, not so much, seems to work fine.

....all the while keeping in mind, it should be a 'holistic' document, covering all version of amiberry past, up to the present.. ;) 

As said, I _like_ doing docs....and if I can write this much about what I feel to be missing, then you've got some idea of how I would revise this page's content.... in not so many words of course =)

So my question is ... do you wish to 'promote' savestate usage with whdload files as explained (with usage tips), while at the same time indicating usage with HDD files may work, depending on actual scenario, to distance that from the emulation usage cases wherein Savestates will definitely fail most all of the time (CD32/CDTV/JIT/PPC/RTG)?

TIA

**Comment by midwan** [comment]
Yes, sure - WHDLoad save states have been tested for years now, and I haven't heard of any problems. So, we know that at least those scenarios work reliably.
The wiki text should be updated to reflect the latest changes, of course. Feel free to help with that :)

**Comment by giantclambake** [comment]
Excellent, leave it with me ... I notice this opens further layer ...hoisting another ticket...
